src/ui/castletiledmap.pas

```diff
@@ -791,7 +794,7 @@ procedure TTiledMap.LoadTMXFile(AURL: string);
 begin
   Doc := nil;
   try
-    ReadXMLFile(Doc, URIDeleteProtocol(AURL));  //todo: what is proper way to open files? (AbsoluteURI ?)
+    ReadXMLFile(Doc, Download(AURL));
```

Comment:
@Kagamma , @tsr8 : Note that this line makes a memory leak, as "Download(AURL)" returns a new stream instance that is nowhere freed.

The easiest solution is to use "URLReadXML(Doc, AURL)". It is defined in CastleXMLUtils, and will free the stream as necessary. It's implementation (copied from CastleXMLUtils sources) is below:

~~~~
procedure URLReadXML(out Doc: TXMLDocument; const URL: String);
var
  Stream: TStream;
begin
  Doc := nil; // clean "out" param at start, just like ReadXMLFile
  Stream := Download(URL, []);
  try
    ReadXMLFile(Doc, Stream);
  finally FreeAndNil(Stream) end;
end;
~~~~